src/layouts/BasicLayout.tsx

```diff
@@ -9,7 +11,9 @@ interface PageProps {
 const BasicLayout: React.FC<PageProps> = (props: PageProps) => {
   return (
     <Fragment>
-      <GlobalStore.Provider>{props.children}</GlobalStore.Provider>
+      <GlobalStore.Provider>
+        <TimeQuerierStore.Provider>{props.children}</TimeQuerierStore.Provider>
+      </GlobalStore.Provider>
```

Comment:
the worst part about using a structure like this is that when the global store changes everything will re-render below it. So I will give you some time to think about a better way to lay your providers 😉

Reply:
Hey @aquibbaig I read in unstated only where we are using this provider only those component will rerender and I even tested it thats what is happening

Reply:
By using I mean consuming the store so that wont be a problem

Reply:
Yeah mb! For some reason, I mistook the store as a component with its own state. Anyways, that still doesn't invalidate the question that this is a bad way to lay out providers in your app.

Reply:
Is there any alternative to avoid this nesting?